### PR TITLE
Search Bar added.

### PR DESCRIPTION
**Pull request by sayansaha1143**
I added a search filter. Its not really pretty, the UI still needs some work.  #53 
But the functionality works fine when you search by the resource name. 
@rbeene Can you have a look at it? Thanks

### CODE REVIEW

**Comment by Khoawala** [comment]
The search function is creating conflict between the category filter and search due to both functions managing the filtered_json. Currently each functions are resetting the list independently based on its criteria.

We should probably discuss whether the search bar or the category filter should be the dominant filter for the result list. Or we could also have one result list that constantly update based on the filter criteria and should only reset when the search is empty and the category list reset.

**Comment by ndesai06** [comment]
Is there a reason why either the search feature or the category list has to be dominant filter for our data? They should be able to work in tandem.
Ex: I filter through the data by selecting only youth/adolescence and childhood support. Once only the resources that fit under those two categories are showing, I can then use the search feature to narrow down the resources even more.

**Comment by sayansaha1143** [comment]
Currently both search filter and category filter are independent of one another. If you filter a certain category and then try to search by name it will ignore the category and search from the full list. I can fix this after the multiple category is merged with the Master to avoid conflicts.

**Comment by rbeene** [comment]
I agree @ndesai06. I expect most users would expect that the filters would work in concert with each other. One other thing I noted was that this is doing a substring search vs a fuzzy search making typos problematic. Did you have a chance to look at that fuzzy search library by chance?

**Comment by sayansaha1143** [comment]
@ndesai06 Yes you are right @Khoawala is working on that. Once that is done and merged I can fix mine and create a new request. @rbeene I havent had a chance to go through the fuzzy search but I can check it and see how that works.

File: src/App.js [hunk 1]
```diff
@@ -123,7 +130,7 @@ class App extends Component {
   }
 
   componentDidMount() {
-    this.callSheets("");
+    this.callSheets("", "");
```
Comment (galiat) [inline src/App.js]: prob a dumb question, but why did you need to change this?

File: src/App.js [hunk 2]
```diff
@@ -77,11 +78,17 @@ class App extends Component {
...
-        else
+        if (selected.length > 0 && type == "name") {
+          var filtered_json = JSON.parse(my_json).filter(function (i) {
+            return i.name.toLowerCase().match( selected.toLowerCase() );
+          });
+          console.log(filtered_json);
+        }
+        else if(selected == "" || selected == "All")
+          var filtered_json = data; 
+        else if(type == "cat")
```
Comment (galiat) [inline src/App.js]: could we just use an `else` here?

**Comment by sayansaha1143** [comment]
Search filter working with multiple category. UI conflicts caused due to connecting sides merge is also fixed. This should be good to be reviewed and merged now. I am working on improving the search bar UI design and also on fuzzy search to see how it can improve the search functionality.
